### ISSUE TITLE
In microsoft Edge browser, loginPopup fails. login popup doesn't show up on the screen. 

### ISSUE THREAD

**Issue by lovin123** [AzureAD/microsoft-authentication-library-for-js#6171]
### Core Library

MSAL.js (@azure/msal-browser)

### Core Library Version

2.14.1

### Wrapper Library

None

### Wrapper Library Version

None

### Public or Confidential Client?

Public

### Description

In microsoft Edge browser, loginPopup fails. login popup doesn't show up on the screen. 

### Error Message

popup_window_error: Error opening popup window. This can happen if you are using IE or if popups are blocked in the browser. Details: BrowserAuthError: empty_window_error: window.open returned null or undefined window object.
![image](https://github.com/AzureAD/microsoft-authentication-library-for-js/assets/95518107/11ebfa09-82c0-470e-a821-597bf13ccef2)

### Msal Logs

[Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - BrowserCrypto: modern crypto interface available
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - BrowserCrypto: modern crypto interface available
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - handleRedirectPromise called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - getAllAccounts called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - BrowserCacheManager.getAccountKeys - No account keys found
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - handleRedirectPromise has been called for the first time, storing the promise
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [b51ec955-e7fb-4fde-850f-090dc7c052c1] : msal.js.browser@2.37.0 : Verbose - initializeServerTelemetryManager called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : @azure/msal-browser@2.37.0 : Verbose - loginPopup called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : @azure/msal-browser@2.37.0 : Verbose - acquireTokenPopup called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - preflightBrowserEnvironmentCheck started
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - preflightInteractiveRequest called, validating app environment
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - getAllAccounts called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : @azure/msal-browser@2.37.0 : Verbose - BrowserCacheManager.getAccountKeys - No account keys found
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - asyncPopup set to false, opening popup before acquiring token
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Default popup window width used. Window width not configured or invalid.
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Default popup window height used. Window height not configured or invalid.
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Default popup window top position used. Window top not configured or invalid.
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Default popup window left position used. Window left not configured or invalid.
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - acquireTokenPopupAsync called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - initializeServerTelemetryManager called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - initializeAuthorizationRequest called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - getRedirectUri called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Initializing BaseAuthRequest
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Authentication Scheme wasn't explicitly set in request, defaulting to "Bearer" request
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - initializeAuthorizationRequest called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - getClientConfiguration called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - getDiscoveredAuthority called
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Creating discovered authority with configured authority
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Attempting to get cloud discovery metadata in the config
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Did not find cloud discovery metadata in the config... Attempting to get cloud discovery metadata from the cache.
instrument.js:111 [Wed, 28 Jun 2023 10:49:19 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Did not find cloud discovery metadata in the cache... Attempting to get cloud discovery metadata from the network.
instrument.js:111 [Wed, 28 Jun 2023 10:49:20 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Attempting to find a match between the developer's authority and the CloudInstanceDiscoveryMetadata returned from the network request.
instrument.js:111 [Wed, 28 Jun 2023 10:49:20 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - cloud discovery metadata was successfully returned from getCloudDiscoveryMetadataFromNetwork()
instrument.js:111 [Wed, 28 Jun 2023 10:49:20 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Auth code client created
instrument.js:111 [Wed, 28 Jun 2023 10:49:20 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - PopupHandler.monitorPopupForHash - polling started
instrument.js:111 [Wed, 28 Jun 2023 10:49:23 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - PopupHandler.monitorPopupForHash - popup window is on same origin as caller
instrument.js:111 [Wed, 28 Jun 2023 10:49:23 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - PopupHandler.monitorPopupForHash - found hash in url
instrument.js:111 [Wed, 28 Jun 2023 10:49:23 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - PopupHandler.monitorPopupForHash - hash contains known properties, returning.
instrument.js:111 [Wed, 28 Jun 2023 10:49:23 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - validateAndExtractStateFromHash called
instrument.js:111 [Wed, 28 Jun 2023 10:49:23 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - Returning state from hash
instrument.js:111 [Wed, 28 Jun 2023 10:49:23 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : msal.js.browser@2.37.0 : Verbose - InteractionHandler.handleCodeResponse called
instrument.js:111 [Wed, 28 Jun 2023 10:49:24 GMT] : @azure/msal-browser@2.37.0 : Verbose - BrowserCacheManager.getAccountKeys - No account keys found
instrument.js:111 [Wed, 28 Jun 2023 10:49:24 GMT] : @azure/msal-browser@2.37.0 : Verbose - BrowserCacheManager.addAccountKeyToMap account key added
instrument.js:111 [Wed, 28 Jun 2023 10:49:24 GMT] : @azure/msal-browser@2.37.0 : Verbose - BrowserCacheManager.getTokenKeys - No token keys found
instrument.js:111 [Wed, 28 Jun 2023 10:49:24 GMT] : @azure/msal-browser@2.37.0 : Verbose - getAllAccounts called
instrument.js:111 [Wed, 28 Jun 2023 10:49:24 GMT] : [d7143fd3-0e84-4ee4-a91d-b59fd71b619b] : @azure/msal-browser@2.37.0 : Verbose - PerformanceClient: Emitting performance events

### MSAL Configuration

```javascript
const msalConfig = {
      auth: {
        clientId:"{}",
        redirectUri: "https://{}"
      },
      system: {
        loggerOptions: {
          loggerCallback: (level: any, message: any, containsPii: any) => {
            if (containsPii) {
              return;
            }
            switch (level) {
              case msal.LogLevel.Error:
                console.error(message);
                break;
              case msal.LogLevel.Warning:
                console.warn(message);
                break;
              case msal.LogLevel.Info:
                console.log(message);
                break;
              case msal.LogLevel.Verbose:
                console.debug(message);
                break;
              default:
                console.log(message);
            }
          },
          logLevel: msal.LogLevel.Verbose
        }
      }
    };
const msalInstance = new msal.PublicClientApplication(msalConfig);
```

### Relevant Code Snippets

```javascript
const msalInstance = new msal.PublicClientApplication(msalConfig);

    msalInstance
      .handleRedirectPromise()
      .then((response: any) => {
        if (response && response.account) {
          acquireToken(response.account);
        } else {
          msalInstance
            .loginPopup()
            .then((loginResponse: any) => {
              if (loginResponse && loginResponse.account) {
                acquireToken(loginResponse.account);
              }
            })
            .catch((error: any) => {
              console.error("Error during login:", error);
            });
        }
      })
      .catch((error: any) => {
        console.error("Error during handleRedirectPromise:", error);
      });
  };

  const acquireToken = (account: any) => {
    const msalInstance = new msal.PublicClientApplication({
      auth: {
        clientId:"{}",
        redirectUri: "https://{}"
      }
    });
    msalInstance.setActiveAccount(account);

    msalInstance
      .acquireTokenPopup({
        scopes: ["Files.Read"]
      })
      .then((response: any) => {
        const OneDriveAccessToken = response.accessToken;
        sessionStorage.setItem("OneDriveAccessToken", OneDriveAccessToken);
        initializePicker(OneDriveAccessToken);
      })
      .catch((error: any) => {
        console.error("Error during token acquisition:", error);
      });
  };
```

### Reproduction Steps

const msalInstance = new msal.PublicClientApplication(msalConfig);

### Expected Behavior

login popup should open even if popups are blocked in browser settings

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

Edge

### Source

Internal (Microsoft)

**Comment by jo-arroyo** [comment]
@lovin123 In order for the popup not to be blocked in this case, the popup needs to be the result of user interaction (e.g. clicking a button). Please give that a try, and take a look at [our samples](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-browser-samples) for examples of how to do this.

**Comment by lovin123** [comment]
@jo-arroyo , thanks for the clarification. Popup works fine when it is implemented as a result of user interaction. I think it would be better if this is included in OneDrive FilePicker 7.2 documentation with some examples showing how to open file picker popup as a result of button click

**Comment by jo-arroyo** [comment]
@lovin123 Glad that helped. I will look into updating our FAQ regarding this. Please [contact OneDrive directly](https://support.microsoft.com/en-us/onedrive) for updates to their documentation.

Closing as issue was addressed.
